packages/ui5-task-compileless/lib/compileLess.js

```diff
@@ -0,0 +1,100 @@
+const log = require("@ui5/logger").getLogger("builder:customtask:compileless");
+const less = require("less-openui5");
+const { ReaderCollectionPrioritized, DuplexCollection, fsInterface } = require("@ui5/fs");
+const Memory = require("@ui5/fs").adapters.Memory
+const FileSystem = require("@ui5/fs").adapters.FileSystem;
+const resourceFactory = require("@ui5/fs/lib/resourceFactory");
+const minimatch = require("minimatch");
+
+/**
+ * Custom task to compile less files in the app folder
+ *
+ * @param {Array} lessResources Array of objects containing the less resource and the corresponding project resource path
+ * @param {fsInterface} fs Node.js styled file system interface
+ * @param {boolean} isDebug Flag indicating debug mode
+ * @returns {Promise<Array>} Promise resolving with the created css resources
+ * @private
+ */
+function compileLess(lessResources, fs, isDebug) {
+    const lessBuilder = new less.Builder({fs});
+
+    return Promise.all(
```

Comment:
if there's async work going on here, you might need to declare `compliLess` as `async` function - if not, `Promise.all` has no effect here

Reply:
done

Reply:
IMO we don't need to declare it as async as we don't use await here and I would have no effect. lessResources.map returns an Arrary of Promises and with Promise.all we "synchronize" the async compiling of the resources and resolve to an array of these resources. But anyway the async keyword does not harm here.